agrirouter-sdk-java-tests/src/test/java/com/dke/data/agrirouter/test/messaging/rest/SendChunkedMessageTest.java

```diff
@@ -0,0 +1,130 @@
+import com.dke.data.agrirouter.impl.messaging.SequenceNumberService;
+import com.dke.data.agrirouter.impl.messaging.encoding.DecodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.encoding.EncodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.rest.FetchMessageServiceImpl;
+import com.dke.data.agrirouter.impl.messaging.rest.SendMessageServiceImpl;
+import com.dke.data.agrirouter.test.AbstractIntegrationTest;
+import com.dke.data.agrirouter.test.Assertions;
+import com.dke.data.agrirouter.test.OnboardingResponseRepository;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Test case to show the behavior for chunked message sending. */
+class SendChunkedMessageTest extends AbstractIntegrationTest {
+
+  private static final int MAX_CHUNK_SIZE = 1024000;
+
+  @ParameterizedTest
+  @MethodSource("fakeRawMessageContentThatHasToBeChunked")
+  void
+      givenLargeContentMessageWhenSendingTheMessageToTheAgrirouterTheSdkShouldHelpToSendTheFileInMultipleChunks(
+          ByteString messageContent, int expectedNrOfChunks)
+          throws IOException, InterruptedException {
+
+    final EncodeMessageService encodeMessageService = new EncodeMessageServiceImpl();
+    final SendMessageServiceImpl sendMessageService = new SendMessageServiceImpl();
+    final OnboardingResponse onboardingResponse =
+        OnboardingResponseRepository.read(OnboardingResponseRepository.Identifier.FARMING_SOFTWARE);
+
+    MessageHeaderParameters messageHeaderParameters = new MessageHeaderParameters();
+    messageHeaderParameters.setTechnicalMessageType(ContentMessageType.ISO_11783_TASKDATA_ZIP);
+    messageHeaderParameters.setApplicationMessageId(MessageIdService.generateMessageId());
+    messageHeaderParameters.setApplicationMessageSeqNo(
+        SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+    messageHeaderParameters.setMode(Request.RequestEnvelope.Mode.PUBLISH);
+
+    PayloadParameters payloadParameters = new PayloadParameters();
+    payloadParameters.setValue(messageContent);
+    payloadParameters.setTypeUrl(SystemMessageType.EMPTY.getKey());
+
+    List<MessageParameterTuple> tuples =
+        encodeMessageService.chunkAndEncode(
+            messageHeaderParameters, payloadParameters, onboardingResponse);
+
+    tuples.forEach(
+        messageParameterTuple ->
+            Assertions.assertTrue(
+                Objects.requireNonNull(messageParameterTuple.getPayloadParameters().getValue())
+                        .toStringUtf8()
+                        .length()
+                    < MAX_CHUNK_SIZE));
```

Comment:
Shouldn't this be <=MAX_CHUNK_SIZE?

quote from PayloadParameters.kt
"* - The maximum body size is equivalent of 1024000 characters/signs"

Reply:
Both is fine. Since the Base64 encoding will always be less than the max chunk size, this is fine. But I changed it to have it clearer.